go/cmd/config/hasura_handler.go

```diff
@@ -162,6 +163,29 @@ func (h *HasuraHandler) handler(w http.ResponseWriter, r *http.Request) {
 		path := fmt.Sprintf("/api/v1/rules/%s/%s", request.Input.Namespace, request.Input.RuleGroupName)
 		httpresp, err := h.cortexQuery(request.Input.TenantID, "DELETE", path, "")
 		response = actions.ToDeleteResponse("Rule group", httpresp, err)
+
+	case "validateTenantName":
+		var request actions.ValidateTenantNamePayload
+		err = json.Unmarshal(reqbody, &request)
+		if err != nil {
+			writeGraphQLError(w, "invalid request payload")
+			return
+		}
+		regex := "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
+		reg, _ := regexp.Compile(regex)
+		invalid_names := request.Input.TenantNames[:0]
+		for _, tenant_name := range request.Input.TenantNames {
+			if !reg.MatchString(tenant_name) {
+				// Append desired values to slice
+				invalid_names = append(invalid_names, tenant_name)
+			}
+		}
+
+		if len(invalid_names) > 0 {
+			error_message := fmt.Sprintf("Invalid tenant names: %v.\nThey must match %v", invalid_names, regex)
+			writeGraphQLError(w, error_message)
+			return
+		}
```

Comment:
How to I return the equivalent of a `200 ok` response here?

Reply:
https://hasura.io/docs/latest/graphql/core/actions/action-handlers.html#returning-a-success-response says

> To return a success response, you must send back a response payload of action’s response type. 

I don't understand this sentence.

It also says

> The HTTP status code must be 2xx for a successful response.

Maybe it's enough to send a 200 response with an empty body?

Probably not.

Further down in these docs they say

> To return a success response, you must send the response of the action’s output type (in this case, UserInfo) with a status code 2xx

that example there might be enlightening.